components/zcash/rs/librustzcash/orchard.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// https://github.com/zcash/librustzcash/blob/zcash_primitives-0.15.0/zcash_primitives/src/transaction/components/orchard.rs
```

Comment:
Can you share something about the motivation here? I really don't like this kind of cut-and-paste vendoring because it's hard to update. Or really, hard to remember it needs updating, since it's not visible to normal advisory notification tools.

Is this an attempt to strip the (large) dependency list of the published `zcash_primitives` and `zcash_encoding` crates? Has the code been customized beyond trimming it? I'd just like to clear about the maintenance and review tradeoffs we're making here.

Reply:
@kdenhartog iirc you've looked at the zcash crates. I'm curious how you see this approach?